src/SqlBinding/TriggerBinding/SqlTableWatcher.cs

```diff
@@ -0,0 +1,505 @@
+                    var acquireLeaseCommand = new SqlCommand(BuildAcquireLeaseOnRowString(row), connection, transaction);
+                    await acquireLeaseCommand.ExecuteNonQueryAsync();
+                }
+
+                await transaction.CommitAsync();
+            }
+        }
+
+        /// <summary>
+        /// Renews the leases held on _rows
+        /// </summary>
+        /// <returns></returns>
+        private async Task RenewLeasesAsync()
+        {
+            using (var connection = SqlBindingUtilities.BuildConnection(_connectionStringSetting, _configuration))
+            {
+                await connection.OpenAsync();
+                // TODO: Set up locks on transaction
+                SqlTransaction transaction = connection.BeginTransaction();
+                foreach (var row in _rows)
+                {
+                    // Not great that we're doing a SqlCommand per row, should batch this
+                    var acquireLeaseCommand = new SqlCommand(BuildRenewLeaseOnRowString(row), connection, transaction);
+                    await acquireLeaseCommand.ExecuteNonQueryAsync();
+                }
+                await transaction.CommitAsync();
+            }
+        }
+
+        /// <summary>
+        /// Releases the leases held on _rows
+        /// </summary>
+        /// <returns></returns>
+        private async Task ReleaseLeasesAsync()
+        {
+            // Don't want to change the _rows while another thread is attempting to renew leases on them
+            await _renewLeasesLock.WaitAsync();
+            try
+            {
+                using (var connection = SqlBindingUtilities.BuildConnection(_connectionStringSetting, _configuration))
+                {
+                    await connection.OpenAsync();
+                    // TODO: Set up locks on transaction
+                    SqlTransaction transaction = connection.BeginTransaction();
+                    foreach (var row in _rows)
+                    {
+                        // Not great that we're doing a SqlCommand per row, should batch this
+                        var releaseLeaseCommand = new SqlCommand(BuildReleaseLeaseOnRowString(row), connection, transaction);
+                        await releaseLeaseCommand.ExecuteNonQueryAsync();
+                    }
+                    await transaction.CommitAsync();
+                }
+                _rows.Clear();
+                _whereChecksOfRows.Clear();
+                _primaryKeyValuesOfRows.Clear();
+            }
+            finally
+            {
+                _renewLeasesLock.Release();
+            }
+        }
+
+        /// <summary>
+        /// Builds the query to create the worker table if one does not already exist (<see cref="CreateWorkerTableAsync"/>)
+        /// </summary>
+        /// <returns>The query</returns>
+        private async Task<string> BuildCreateTableCommandStringAsync()
+        {
+            await GetPrimaryKeysAsync();
```

Comment:
I thought about this as well. I couldn't find anything in the Change Tracking docs about this, but when I tried to change the primary key constraint on one of my tables by first deleting it and then creating another constraint, Change Tracking prevented me from deleting it. From what I understand, this is the only way to mess with primary key constraints since it doesn't seem like you can add another primary key constraint after one already exists. That's what [this doc](https://docs.microsoft.com/en-us/sql/relational-databases/tables/modify-primary-keys?view=sql-server-ver15#TsqlProcedure) made it seem like at least.